DRM Wizard

elken opened the pull request:
- Added a wizard to find DRM games (uplay, origin & steam atm)
- Shrunk Library ctor
- Few minor changes elsewhere

NOTE: Until I find out where they're stored, user-configured launch options don't carry over.

EDIT: Ok I just found the file, hope is yet.

Commit:
Skeleton work for QWizard init stuff

- Added icons
- Modified relevant files
- Created basic wizard

Commit:
More work on Wizard

- Added basic hardcoded Linux steamFind stuff
- Added widget to display the info
- Reorganized file structure

Still need to test on Windows with big steam lib and implement the other find functions. Threading would be a dream too.

Commit:
Wizard stuff working for Steam

- Uncommented broken code, will remove later
- Made findSteam a bit more generic

Need to clean up the ifdefs

Commit:
Merge branch 'updev' into flowTwoPointOh

Commit:
Fixed a build issue and added svg support

Commit:
Missed an ifdef issue

Commit:
Most of the find* functionality is in, just needs testing

- Moved the functionality to the correct page
- Shrunk DRMPage ctor
- Shrunk Library ctor
- Adjusted cmake to copy_if_different

Need to get the game finding stuff to work, and get it multithreaded

Commit:
Adjusted title to suit platform

Commit:
Got multithreaded finding working

- Utilizes std::future to spin off a thread for each find* function
- Added some labels to get the path

A cleaner way to handle path should be found, but this will do for now

Commit:
Threads for Linux

Commit:
Libs fix (Linux)

Commit:
Various Steam changes, some DB changes

- Made `tuple` references into `pair`, we weren't using a `tuple`
- Made *.toInt() query results to *.toUInt()
- Adjusted SteamFind to use `boost::optional` rather than sloppy try-catch statements, finds all names now
- Added `std::sort` for `steamVector`, will implement sorting algorithms for Library soon
- Added `printTree` function to recursively print a ptree

Currently a weird bug with certain games picking up the exe/dir incorrectly, and other DRM libs need to be added

Commit:
Merge branch 'flowTwoPointOh' of https://github.com/elken/Client into flowTwoPointOh

Commit:
Constness

Commit:
Boost fix for Semaphore

Commit:
Added checkboxes

- Some layout changes
- Added checkboxes for Steam
- Added selectAll, deselectAll, invertAll buttons that work
- Set default size

Commit:
Merge branch 'flowTwoPointOh' of https://github.com/elken/Client into flowTwoPointOh

Commit:
Merge branch 'updev' into flowTwoPointOh

- Merged upstream dev
- Finally cleaned up Library properly

elken committed:
Random space, ok

Commit:
Some Steam fixes

- Moved the autofind stuff around to accommodate missing oslists
- Increased width on game title edit box

Commit:
Added include guards for tab labels

Commit:
Missing endif

Commit:
Adding Steam games to database now works

- Added Steam games to db (Some of this code will need to be refactored and made more generic)

Very hacked, but it works for now.

Commit:
Added watcher to autorefresh GameWidget

- Implemented addGames function for GameLists
- Moved GameList typedef to Database
- Added fswatcher for dbfile
- Moved refreshGames to a slot
-

Commit:
I maked a stupid

Commit:
Added Origin games to tab widget

- Added initial call to refreshGames in Library ctor
- Added sloppy layout code for Origin games
- Refactored findOriginGames to use a property tree instead

Commit:
Tweaked findOriginGames to allow import into DB soon

- Moved code about to best stick to logic of whether a DRM item has been found
- Updated title to logically include Origin games (including correct grammar)
-  Added game count to originTree
- Added code to dump the Origin game info to stdout

Some bugs regarding flow, will iron out in future commits.

Commit:
Fixed grammar for titles

- Results title correctly and grammatically reflects results
- Uplay games are found, but not yet added to the layout (next commit)

Commit:
Origin and Uplay games can be added to the db

- Made getGameCount const so it can be used with a const "this"
- Added correct games to db

elken committed:
Final commit, various fixes

- Changed some of the language to make it consistent
- Added somewhat smarter checking for a uplay installation
- Formatting
- Basic sorting
- IgnoreList for std uplay install (to ignore base folders)

lolzballs commented on Source/Main.cpp:
```diff
@@ -1,24 +1,32 @@
+#include <QFile>
+#include <QDebug>
+#include <iostream>
+#include "Wizard.h"
```
This should be moved up to the other local includes.

lolzballs commented on Source/Wizard.cpp:
```diff
@@ -0,0 +1,775 @@
+#include <QtWidgets>
+#include <iostream>
+#include "Wizard.h"
```
These includes should follow the style documented.

elken committed:
Update Main.cpp

lolzballs commented on Source/Wizard.h:
```diff
@@ -0,0 +1,120 @@
+
+#include <QtWidgets>
+#include <vector>
+#include "Database.h"
```
Same thing

elken committed:
Update Wizard.cpp

elken committed:
Update Wizard.h

lolzballs commented:
Any particular reason for not including documentation?

elken commented:
I just wanted it done & looked over. I'll add docs before merge.

Tom-Todd committed:
Testing Settings

Tom-Todd committed:
Testing Settings

Tom-Todd committed:
Initial Settings Widget

-Boiler plate code for settings

Tom-Todd committed:
Settings Widget

-Added documentation

Tom-Todd committed:
Rename settings.ui to Settings.ui

Tom-Todd committed:
Settings Widget

-Fixed missing header

Tom-Todd committed:
Merge branch 'flowTwoPointOh' of github.com:Tom-Todd/Client into flowTwoPointOh

Tom-Todd committed:
Settings Widget

-CMakeLists fixes

Tom-Todd committed:
Settings Widget

-Added settings button to re-run setup wizard on demand
-Added check to see if the database exists

>Will fix issue with database access when re-running the wizard

Tom-Todd committed:
Settings Widget

-Fixed database access issues when re-running wizard

Tom-Todd committed:
Settings Widget

-Fixed issue of Wizard not showing on first launch
-Fixed issue of Wizard not clearing addedVector correctly.

Tom-Todd committed:
Library Widget

-Added sort to refreshGames

Tom-Todd committed:
Settings Widget

-Added option to reset the database

Tom-Todd committed:
Settings Widget

-Fixed database access issue

Tom-Todd committed:
Settings Widget

-Added options to change the three accent colours
-Added option to reset accents to default
-Changed location of palette.ini to appdata

Tom-Todd committed:
Settings Widget

-Fixed Issue which stopped colours being updated correctly

Tom-Todd committed:
Settings Widget

-Restructured settings widget and created placeholder UI areas

Tom-Todd committed:
Settings Widget

-Fix to previous build error (removed placeholder text)

Tom-Todd committed:
Settings Widget

-Removed placeholder text fields

elken committed:
AddGameWizard and DRMSetupWizard docs

- Finished docs for DRMSetupWizard
- Renamed Wizard to DRMSetupWizard
- Added basic AddGameWizard

elken commented:
@lolzballs @Jtramey Docs are in, check it out and when one of you two signs it off I'll merge

elken committed:
Merge branch 'flowTwoPointOh' of https://github.com/Tom-Todd/Client into flowTwoPointOh

elken commented:
Merged #95 into this one

awslattery commented:
Nice work guys.

elken commented on Source/MainPanel.cpp:
```diff
@@ -26,7 +28,7 @@ MainPanel::MainPanel(QWidget* parent)
 void MainPanel::init()
 {
-    p = new QSettings("palette.ini", QSettings::IniFormat);
+    p = new QSettings(QSettings::IniFormat, QSettings::UserScope, "Project Ascension", "palette");
```
@Hal9007
